frontend/public/components/sidebars/resource-sidebar-samples.tsx

```diff
@@ -0,0 +1,284 @@
+        templateName: 'web-allow-production',
+        kind: referenceForModel(NetworkPolicyModel),
+      },
+    ],
+  )
+  .setIn(
+    [referenceForModel(ResourceQuotaModel)],
+    [
+      {
+        title: 'Set compute resource quota',
+        description: 'Limit the total amount of memory and CPU that can be used in a namespace.',
+        templateName: 'rq-compute',
+        kind: referenceForModel(ResourceQuotaModel),
+      },
+      {
+        title: 'Set maximum count for any resource',
+        description:
+          'Restrict maximum count of each resource so users cannot create more than the allotted amount.',
+        templateName: 'rq-counts',
+        kind: referenceForModel(ResourceQuotaModel),
+      },
+      {
+        title: 'Specify resource quotas for a given storage class',
+        description:
+          'Limit the size and number of persistent volume claims that can be created with a storage class.',
+        templateName: 'rq-storageclass',
+        kind: referenceForModel(ResourceQuotaModel),
+      },
+    ],
+  )
+  .setIn(
+    [referenceForModel(RoleModel)],
+    [
+      {
+        title: 'Allow reading the resource in API group',
+        description: 'This "Role" is allowed to read the resource "Pods" in the core API group.',
+        templateName: 'read-pods-within-ns',
+        kind: referenceForModel(RoleModel),
+      },
+      {
+        title: 'Allow reading/writing the resource in API group',
+        description:
+          'This "Role" is allowed to read and write the "Deployments" in both the "extensions" and "apps" API groups.',
+        templateName: 'read-write-deployment-in-ext-and-apps-apis',
+        kind: referenceForModel(RoleModel),
+      },
+      {
+        title: 'Allow different access rights to different types of resource and API groups',
+        description:
+          'This "Role" is allowed to read "Pods" and read/write "Jobs" resources in API groups.',
+        templateName: 'read-pods-and-read-write-jobs',
+        kind: referenceForModel(RoleModel),
+      },
+      {
+        title: 'Allow reading a ConfigMap in a specific namespace (for RoleBinding)',
+        description:
+          'This "Role" is allowed to read a "ConfigMap" named "my-config" (must be bound with a "RoleBinding" to limit to a single "ConfigMap" in a single namespace).',
+        templateName: 'read-configmap-within-ns',
+        kind: referenceForModel(RoleModel),
+      },
+      clusterRoleBinding1,
+      clusterRoleBinding2,
+    ],
+  )
+  .setIn([referenceForModel(ClusterRoleModel)], [clusterRoleBinding1, clusterRoleBinding2]);
+
+export const getResourceSidebarSamples = (kindObj: K8sKind, yamlSampleObj: K8sResourceKind) => {
+  const yamlSampleData = !_.isEmpty(yamlSampleObj)
+    ? _.filter(
+        yamlSampleObj.data,
```

Comment:
This doesn't look like a `K8sResourceKind` if it has a `data` prop. I think you want `yamlSampleObj: FirehoseResult`